lcls_tools/superconducting/sc_magnet.py

```diff
@@ -0,0 +1,76 @@
+from typing import Optional, TYPE_CHECKING
+
+from lcls_tools.common.controls.pyepics.utils import PV
+from lcls_tools.superconducting import sc_linac_utils as utils
+
+if TYPE_CHECKING:
+    from lcls_tools.superconducting.sc_cryomodule import Cryomodule
+
+
+class Magnet(utils.SCLinacObject):
+    """
```

Comment:
I'm looking at his magnet class now, but it looks like it makes PV objects on creation? Part of the problem with our applications was that all of those connections were significantly impacting launch times so I wrote them as properties that only get created when that specific PV is called. Am I reading it correctly?

Reply:
I think that's only an issue if you create the magnet for use (i.e. make a beam line of devices object). Is there a case where you want to initialize a magnet but not use it? Maybe @MattKing06 can comment if I'm missing something.

Reply:
@lisazacarias @nneveu  When the `devices.magnet` object is created for an area/beampath. It attempts to connect to the PVs on creation. 

It could be a potential feature to only connect on use and to periodically poll the connection status of the PV object to ensure the user is warned when a PV becomes disconnected from the controls network.

Reply:
@nneveu Yeah I actually intentionally wrote the sc linac classes so that people don't make their own orphan devices. If you want a magnet (or a cavity, or a piezo, etc), you need something like `machine.linacs[0].cryomodules[1].quad` and all objects in the linac are created when you create a machine. It was defensive programming for the SRF machine lol

Reply:
But do we need to create all devices if they are not being used? Especially for the magnet case, there are lots of operations we do w/ magnets that are not dependent on other functionality